pkg/virt-handler/ratelimitcmd/backoff_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package ratelimitcmd_test
```

Comment:
That on is a `flowcontrol` rate limiter, no idea what it is.. but it is for sure unrelated with an executor/cmd rate limiter.

Reply:
I"m not saying the `flowcontrol` should be used. Just having the new ratelimiter under the same package.

Reply:
I think `ratelimitercmd` and `pkg/util/ratelimiter` even tough resembles each other semantically, they are totally two different things.
But I do agree `ratelimitercmd` should be moved to pkg/util/,
what you think about moving there as is a side to the one that exists?

Reply:
Yes, that's what I meant.

Reply:
I do not think anything should be under something called `util`, and there is nothing in common about `ratelimiter` as a super-package. Multiple objects may have a ratelimiter behavior, in this case it is a cmd/exec.

How about creating an `executor` package, which has a ratelimter behavior?
IMO it should have been the same with the other one that exists now, it is a "flowcontrol" thing that has a ratelimiter option and not the other way around.

Reply:
I think the `ratelimiter` can leave by its own and it not only for the `executor`. The `executor` is just one usage of the `ratelimiter`. But I won't block the PR on it.
@ormergi what do you think?

Reply:
Thinking of this again, I do agree we should not put it under pkg/util (we had bad experience with util packages..),
the current `ratelimiter` package we have (the one that warping `flowcontrol`) should have different name this is what confusing.
If there were more usages for the rate-limiter part (basically backoff.go) of the executor 
it would have been natural to put it under its own package `pkg/ratelimiter` for reuse.
But since there are not other consumers at the moment we can keep it all under one package `executor`.

If you prefer to have the rate-limiter part (`backoff.go`) on its own package for viability and encouraging others to use it
we can split the it to two packages `pkg/ratelimiter` and `pkg/executor`, and dill with `pkg/util/ratelimiter` later

`pkg/`:
|_ `executor`
|   |_ pool.go
|   |_ executor.go
|    ...
|_ `ratelimiter`
    |_ backoff.go
@AlonaKaplan  @EdDev  WDYT?

Reply:
I think it is a property of an executor at the moment. I would only promote it to its own package if another user of it appears.

> I think the `ratelimiter` can live by its own and it not only for the `executor`

Yes, but it is still a property of the executor and having its own package is just an option if we see it used by another functionality. I do not have one in mind at the moment.
And I also do not like `util` or have a name that this one can live nicely with the other one you mentioned.